packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -425,19 +453,34 @@ export class BaseSnapExecutor {
     };
 
     // To harden and limit access to internals, we use a proxy.
-    const proxy = new Proxy(provider, {
-      get(target, prop: keyof StreamProvider) {
-        if (prop === 'request') {
-          return request;
-        } else if (['on', 'removeListener'].includes(prop)) {
-          return target[prop];
-        }
-
-        return undefined;
+    // Proxy target is intentionally set to be an empty object, to ensure
+    // that access to the prototype chain is not possible.
+    const proxy = new Proxy(
```

Comment:
This seems to be the same as the proxy above, so can we merge this into a function to reduce code duplication?

Reply:
Yes, good catch. For now it is the same thing, I extracted it to utility function and refactored.